tests/utils/test_rest_utils.py

```diff
@@ -93,3 +94,15 @@ def confirm_request_params(**kwargs):
     response = rest_utils.databricks_api_request('clusters/list', 'PUT',
                                                  req_body_json={'a': 'b'}, params='x=y')
     assert response == {'OK': 'woo'}
+
+
+def test_json_dumps_default_numpy_int64():
+    test_number = numpy.int64(42)
+    defaulted_val = rest_utils.default(test_number)
+    assert defaulted_val is 42
+
+
+def test_json_dumps_default_non_serializable():
+    test_number = numpy.int32(42)
```

Comment:
Looks OK to me, but are we sure we don't want to convert int32 to int as well? It seems that we should do that and create another test case for something that fails (just create your own class in the test or something).

Reply:
Interesting, apparently `numpy.int32` is not serialisable by default in Python3. And I did a quick test on float: `numpy.float16` is not serialisable but `numpy.float32` is serialisable. 

If we're going to fix this, then we'll need to test for every numpy data structure. 
Moreover, according to [Python issue 24313](https://bugs.python.org/issue24313), the behaviour is different according to the operating system. 

Should we warn users to return `prediction` in serialisable format instead of auto-convert their numpy data structure?

Reply:
I don't know if it's a complete list, but I guess the 'enumerate-all-types' solution is proposed here:
https://github.com/mpld3/mpld3/issues/434#issuecomment-340255689

If we ask users to return prediction in a serializable format, it would still be good to let them know _how_ to do that properly given a numpy array (which is a common data structure to start with). Doing so seems about as hard as implementing the conversion ourselves.

Reply:
I think it's OK to just start with a few common types. People are not that likely to be using the weirder ones (e.g. less than 32 bits precision). Also I don't think it hurts to list types that happen to be natively serializable.. the code here just won't run.